File: generate-random-password.py
import random
import string
import argparse

def generate_password(length=12, include_digits=True, include_special_chars=True):
    characters = string.ascii_letters
    if include_digits:
        characters += string.digits
    if include_special_chars:
        characters += string.punctuation

    password = ''.join(random.choice(characters) for _ in range(length))
    return password

def main():
    parser = argparse.ArgumentParser(description="Generate a random password.")
    parser.add_argument("--length", type=int, default=12, help="Length of the password")
    parser.add_argument("--no-digits", action="store_false", dest="include_digits", help="Exclude digits")
    parser.add_argument("--no-special-chars", action="store_false", dest="include_special_chars", help="Exclude special characters")

    args = parser.parse_args()

    password = generate_password(args.length, args.include_digits, args.include_special_chars)
    print("Generated Password:", password)

if __name__ == "__main__":
    main()

File: generate-random-password.go
package main

import (
	"flag"
	"fmt"
	"math/rand"
	"time"
)

const (
	letterBytes        = "abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ"
	digitBytes         = "0123456789"
	specialCharBytes   = "!@#$%^&*()-_=+[]{}|;:'\",.<>?/"
	defaultPasswordLen = 12
)

var (
	length              int
	includeDigits       bool
	includeSpecialChars bool
)

func init() {
	flag.IntVar(&length, "length", defaultPasswordLen, "Length of the password")
	flag.BoolVar(&includeDigits, "include-digits", true, "Include digits in the password")
	flag.BoolVar(&includeSpecialChars, "include-special-chars", true, "Include special characters in the password")
	flag.Parse()

	rand.Seed(time.Now().UnixNano())
}

func generatePassword(length int, includeDigits bool, includeSpecialChars bool) string {
	characters := letterBytes
	if includeDigits {
		characters += digitBytes
	}
	if includeSpecialChars {
		characters += specialCharBytes
	}

	password := make([]byte, length)
	for i := range password {
		password[i] = characters[rand.Intn(len(characters))]
	}

	return string(password)
}

func main() {
	password := generatePassword(length, includeDigits, includeSpecialChars)
	fmt.Println("Generated Password:", password)
}
